pallets/circuit/src/lib.rs

```diff
@@ -248,7 +268,36 @@ pub mod pallet {
         // dispatched.
         //
         // This function must return the weight consumed by `on_initialize` and `on_finalize`.
-        fn on_initialize(_n: T::BlockNumber) -> Weight {
+        fn on_initialize(n: T::BlockNumber) -> Weight {
+            // Check every XtxTimeoutCheckInterval blocks
+            if n % T::XtxTimeoutCheckInterval::get() == T::BlockNumber::from(0u8) {
+                // Go over all unfinished Xtx to find those that timed out
+                <ActiveXExecSignalsTimingLinks<T>>::iter()
```

Comment:
Is there a chance that this could block other execution if there are too many of these?

I think we may need to do something like what pallet contracts does for the  Deletion Queue which is essentially take a slice of this as it's looping.

We should also ensure the weights for this is calculated properly as it could be a bug in the future.

Or at the very least, make some tech debt to do it later :)

Reply:
this is only looped over here in oninitialize - iterating over this linked map should be much faster already than iterating over the full xtx structures. Will take a look at deletion queue